### PR TITLE
chore(node): prettify CoreAccess component construction

### PR DESCRIPTION
**Pull request by Wondertan**

### CODE REVIEW

File: node/components.go [hunk 1]
```diff
@@ -71,21 +72,7 @@ func baseComponents(cfg *Config, store Store) fxutil.Option {
 		p2p.Components(cfg.P2P),
 		// state components
 		fxutil.ProvideIf(cfg.Core.RemoteAddr != "", state.NewService),
-		fxutil.ProvideIf(cfg.Core.RemoteAddr != "", func(lc fx.Lifecycle) (state.Accessor, error) {
-			ks, err := store.Keystore()
```
Comment (Wondertan) [inline node/components.go:75]: It is provided to DI so we don't need to extract here.

File: node/components.go [hunk 2]
```diff
@@ -71,21 +72,7 @@ func baseComponents(cfg *Config, store Store) fxutil.Option {
...
-			if err != nil {
-				return nil, err
-			}
-			lc.Append(fx.Hook{
-				OnStart: ca.Start,
-				OnStop:  ca.Stop,
-			})
-			return ca, nil
-		}),
+		fxutil.ProvideIf(cfg.Core.RemoteAddr != "", statecomponents.CoreAccessor(cfg.Core.RemoteAddr)),
```
Comment (Wondertan) [inline node/components.go:75]: Making this a one-liner, like all the other components.

**Review by renaynay — APPROVED** [review]
LGTM - I made some of these changes already [here](https://github.com/celestiaorg/celestia-node/pull/573/files#diff-febd9ec62adaca7eed5a6e23a67d1c22735462c3253f3c3b5f6e9098ead8de26R74) but not making it a one-liner so let's merge this quickly and i'll rebase mine on yours.

**Comment by Wondertan** [comment]
Hmm, don't see similar changes there, only changes on the same line.

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**